src/functions-and-arrays.js

```diff
@@ -1,41 +1,128 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  if ( a >= b ) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(array) {
+  let longest;
+  if ( array.length === 0 ) { 
+    return null
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      const word = array[i];
+      if ( !longest || longest.length < word.length ) {
+        longest = word;
+      }
+    }
+    return longest;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return 0;
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      total += array[i]
+    } 
+  }
+  return total;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return 0;
+    } else {
+      for ( let i = 0; i < array.length; i++ ) {
+        const item = array[i]
+        if ( typeof item === "string" ) {
+          total += item.length;
+        } else if ( typeof item === "number" ) {
+          total += item
+        } else if ( typeof item === "boolean" ) {
+          total += item
```

Comment:
```suggestion
        } else if ( typeof item === "number" || typeof item === "boolean" ) {
          total += item
        } 
```
Lo que has puesto esta perfecto!!! pero sabes que? los booleanos tambien se pueden sumar, es decir, true es igual a 1 y false es igual a 0, asi que puedes ahorrarte un else if :D